sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/SelectQuery.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using Azure.DigitalTwins.Core.Queries.QueryBuilders;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.DigitalTwins.Core.QueryBuilder
+{
+    /// <summary>
+    /// Query object without any clauses.
+    /// </summary>
+    public sealed class SelectQuery : QueryBase
+    {
+        private readonly FromQuery _delegationClause;
+        private readonly AdtQueryBuilder _parent;
+        private SelectClause _clause;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SelectQuery"/> class.
+        /// </summary>
+        internal SelectQuery(AdtQueryBuilder parent, FromQuery upstreamFromQuery)
+        {
+            _parent = parent;
+            _delegationClause = upstreamFromQuery;
+        }
+
+        /// <summary>
+        /// Used to add a select clause (and its corresponding argument) to the query.
+        /// </summary>
+        /// <param name="args"> The arguments that define what we select (eg. *). </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery Select(params string[] args)
+        {
+            _clause = new SelectClause(args);
+            return _delegationClause;
+        }
+
+        /// <summary>
+        /// Used when applying the TOP() aggregate from the ADT query language. Same functionality as select
+        /// but inserts TOP() into the query structure as well.
+        /// </summary>
+        /// <param name="count"> The argument for TOP(), ie the number of instances to return. </param>
+        /// <param name="args"> The arguments that can be optionally passed with top (eg property name). </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectTop(int count, params string[] args)
+        {
+            // turn into correct format -- eg. SELECT TOP(3)
+            StringBuilder topArg = new StringBuilder().Append($"{QueryConstants.Top}({count})").Append(' ');
+
+            // append optional arguments separated by commas
+            topArg.Append(string.Join(", ", args));
+
+            _clause = new SelectClause(new string[] { topArg.ToString() });
+            return _delegationClause;
+        }
+
+        /// <summary>
+        /// Used when applying the COUNT() aggregate from the ADT query language. Similar to SelectTop().
+        /// </summary>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectCount()
+        {
+            string countArg = new StringBuilder().Append($"{QueryConstants.Count}()").Append(' ').ToString();
```

Comment:
Why not just do string interpolation? Doesn't really need the overhead of a StringBuilder.

`string countArg = $"{QueryConstants.Count}() ";`